### PR TITLE
feat(User): username, public profile flag & documents

### PR DESCRIPTION
**Pull request by tpreusse**
affects: @orbiting/backend-modules-auth

BREAKING CHANGE: move arbitrary user data to _raw

Other changes:
- add `Roles.userIsMe` and `Roles.userIsMeOrInRoles` helper
- `email` becomes optional—not always accessible

### CODE REVIEW

File: packages/auth/migrations/sqls/20171217200855-profile-up.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+ALTER TABLE users
+  ADD COLUMN "username"     citext unique,
+  ADD COLUMN "hasPublicProfile"  boolean not null default false
+;
```
Comment (patte) [inline packages/auth/migrations/sqls/20171217200855-profile-up.sql]: Missing: `CREATE EXTENSION IF NOT EXISTS "citext"`

File: packages/auth/migrations/sqls/20171217200855-profile-down.sql [hunk 2]
```diff
@@ -0,0 +1,4 @@
+alter table users
+  drop column "username",
+  drop column "hasPublicProfile"
+;
```
Comment (patte) [inline packages/auth/migrations/sqls/20171217200855-profile-down.sql]: case consistency would be nice.

File: packages/auth/graphql/resolvers/_queries/user.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  if (!id && !username) {
+    return null
+  }
+  const user = await pgdb.public.users.findOne({
+    id,
+    username
+  }, { skipUndefined: true })
+  if (
+    !user ||
+    (!user.hasPublicProfile && Roles.userIsMe(user, me))
```
Comment (patte) [inline packages/auth/graphql/resolvers/_queries/user.js]: should be `(Roles.userIsMe(user, me) || !user.hasPublicProfile)` no?

File: packages/auth/migrations/sqls/20171217200855-profile-up.sql (hunk 1 above)
Comment (tpreusse) [reply packages/auth/migrations/sqls/20171217200855-profile-up.sql]: ✅

**Comment by tpreusse** [comment]
Added all features for the profile. @patte please review again

Tested with locally linked with `publikator-backend`:
```
1..848
# tests 848
# pass  848

# ok

✨  Done in 325.23s.
```
https://github.com/orbiting/publikator-backend/pull/22

**Pull request merged**